internal/pkg/mount/system.go

```diff
@@ -235,6 +237,25 @@ func SystemPartitionMount(ctx context.Context, r runtime.Runtime, logger *log.Lo
 	mountStatus.TypedSpec().FilesystemType = mountpoint.Fstype()
 	mountStatus.TypedSpec().Encrypted = encrypted
 
+	if encrypted {
+		encryptionProviders := make([]string, 0, len(o.Encryption.Keys()))
+
+		for _, cfg := range o.Encryption.Keys() {
+			switch {
+			case cfg.Static() != nil:
+				encryptionProviders = append(encryptionProviders, cfg.Static().String())
+			case cfg.NodeID() != nil:
+				encryptionProviders = append(encryptionProviders, cfg.NodeID().String())
+			case cfg.KMS() != nil:
+				encryptionProviders = append(encryptionProviders, cfg.KMS().String())
+			case cfg.TPM() != nil:
+				encryptionProviders = append(encryptionProviders, cfg.TPM().String())
+			}
+		}
+
+		mountStatus.TypedSpec().EncryptionProviders = maps.Keys(slices.ToSet(encryptionProviders))
```

Comment:
We can directly gather them into a set (`map[string]struct{}{}`) instead of gathering them into a slice first and then doing `slices.ToSet(encryptionProviders)`.

Reply:
fixed, also verified by running reset on ephemeral and verified status changes